### DISCUSSION TITLE
Stream Youtube Videos Directly

### DISCUSSION THREAD

**Discussion by benf2004** [vicwomg/pikaraoke#262] [Ideas]
I would love it if there was a way to stream Youtube directly within the interface instead of downloading, which has been unreliable for me.

**Comment by Jake-pi** [comment]
I had a similar experience, with song downloads taking a long time. Sometimes youtube.dl would need to be updated, but you wouldn't know it for around 10 - 15 minutes. I recently installed PiKaraoke on a Pi5 8GB, and new song downloads only take about 2 minutes now. I think direct streaming would probably be difficult for the developer to do. 

For me, upgrading my Pi hardware significantly improved download times, and made it such that a newly downloaded song will be ready before the current song being played was finished. It hasn't happened yet, but I imagine if a youtube.dl update is needed, it will be apparent quick enough for it to be a minor interruption, rather than a disappointment not being able to download new songs for a while. 

I was also able to install PiKaraoke on a complete Bullseye 64bit installation, including desktop and made a link to start PiKaraoke from the desktop. This makes it so I can use the Pi for media and internet on my TV as well as run PiKaraoke, making better use of the hardware. For me this helped justify the purchase of new hardware. My old Pi4 8gb is now being used for another project. I hope this helps improve your experience.

**Reply by vicwomg** [reply]
2 minutes seems far too long to download a track. Usually takes under 30s. Not sure what's going on there. Might be worth running with debug logging (add `-l10` to the launch command) and observing the output.
